/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.transactionservice;

import "google/protobuf/empty.proto";
import public "models/transaction.proto";

service TransactionService {
  rpc TransactionsCreate (TransactionsCreateRequest) returns (Transaction);

  rpc TransactionsDelete (TransactionsDeleteRequest) returns (google.protobuf.Empty);

  rpc TransactionsReadAll (google.protobuf.Empty) returns (TransactionsReadAllResponse);

  rpc TransactionsReadOne (TransactionsReadOneRequest) returns (Transaction);

  rpc TransactionsUpdate (TransactionsUpdateRequest) returns (Transaction);

}

message TransactionsCreateRequest {
  // Transaction to create
  Transaction transaction = 1;

}

message TransactionsDeleteRequest {
  // ID of the transaction.
  int32 transactionId = 1;

}

message TransactionsReadAllResponse {
  repeated Transaction data = 1;
}

message TransactionsReadOneRequest {
  // ID of the transaction.
  int32 transactionId = 1;

}

message TransactionsUpdateRequest {
  // ID of the transaction.
  int32 transactionId = 1;
  Transaction transaction = 2;

}

